### PR TITLE
[file-saver-es] Add types

### PR DESCRIPTION
**Pull request by BendingBender**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/file-saver-es/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// Project: https://github.com/eligrey/FileSaver.js/
+// Definitions by: Cyril Schumacher <https://github.com/cyrilschumacher>
+//                 Daniel Roth <https://github.com/DaIgeb>
+//                 Chris Barr <https://github.com/chrismbarr>
+//                 HitkoDev <https://github.com/HitkoDev>
+//                 JounQin <https://github.com/JounQin>
+//                 BendingBender <https://github.com/bendingbender>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace saveAs;
```
Comment (JounQin) [inline types/file-saver-es/index.d.ts:11]: Is this really needed? Exposing all symbols into `window` makes tree shaking unavailable.

File: types/file-saver-es/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Type definitions for file-saver-es 2.0
+// Project: https://github.com/eligrey/FileSaver.js/
```
Comment (JounQin) [inline types/file-saver-es/index.d.ts:2]: It's not the real path of the fork project.

File: types/file-saver-es/test/file-saver-es-tests.ts [hunk 3]
```diff
@@ -28,13 +28,13 @@ function testDeprecatedSaveAs() {
  * @summary Test for "saveAs" function on the window object.
  */
 function testWindowSaveAs() {
```
Comment (JounQin) [inline types/file-saver-es/test/file-saver-es-tests.ts:30]: Like above, I don't think testing `window` is needed.

File: types/file-saver/test/file-saver-tests.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import { saveAs, FileSaverOptions } from 'file-saver';
```
Comment (JounQin) [inline types/file-saver/test/file-saver-tests.ts:1]: Why `saveAs` is using `import` which is different from `saveAsReq` here?

File: types/file-saver-es/index.d.ts (hunk 1 above)
Comment (BendingBender) [reply types/file-saver-es/index.d.ts:11]: This is not exposing anything, these are just type definitions. The library is exposing everything into window, nothing I can do about it. But I could lie to users by not mentioning it...

File: types/file-saver-es/index.d.ts (hunk 2 above)
Comment (BendingBender) [reply types/file-saver-es/index.d.ts:2]: Do oyu know th real path?

File: types/file-saver-es/test/file-saver-es-tests.ts (hunk 3 above)
Comment (BendingBender) [reply types/file-saver-es/test/file-saver-es-tests.ts:30]: As I said, nothing I can do about it, the library exposes stuff, type definitions only describe what the library does.

File: types/file-saver/test/file-saver-tests.ts (hunk 4 above)
Comment (BendingBender) [reply types/file-saver/test/file-saver-tests.ts:1]: To test named imports (`import { saveAs } from 'file-saver'`) vs CommonJS style imports (`import saveAs = require('file-saver')`). The former works in ES6 module environments, the latter with node requires.

File: types/file-saver-es/index.d.ts (hunk 2 above)
Comment (BendingBender) [reply types/file-saver-es/index.d.ts:2]: Nvm, found it.

File: types/file-saver-es/test/file-saver-es-tests.ts [hunk 5]
```diff
@@ -28,13 +28,13 @@ function testDeprecatedSaveAs() {
...
-    const data: Blob = new Blob(["Hello, world!"], {type: "text/plain;charset=utf-8"});
+    const data: Blob = new Blob(['Hello, world!'], { type: 'text/plain;charset=utf-8' });
     const filename = 'hello world.txt';
     const options: FileSaverOptions = {
-        autoBom: false
+        autoBom: false,
     };
 
-    window.saveAs(data, filename, options);
+    window.saveAs.saveAs(data, filename, options);
```
Comment (JounQin) [inline types/file-saver-es/test/file-saver-es-tests.ts:37]: Why this is changed like so?

File: types/file-saver-es/test/file-saver-es-tests.umd.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
+/**
```
Comment (JounQin) [inline types/file-saver-es/test/file-saver-es-tests.umd.ts:1]: Is `umd` testing for `es` needed?

File: types/file-saver/test/file-saver-tests.ts (hunk 4 above)
Comment (JounQin) [reply types/file-saver/test/file-saver-tests.ts:1]: `file-saver` is not `ES`, so I don't think we need to test that?

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#esmoduleinteropallowsyntheticdefaultimports

File: types/file-saver-es/test/file-saver-es-tests.umd.ts (hunk 6 above)
Comment (BendingBender) [reply types/file-saver-es/test/file-saver-es-tests.umd.ts:1]: Yes, this package still exposes a UMD bundle.

File: types/file-saver/test/file-saver-tests.ts (hunk 4 above)
Comment (BendingBender) [reply types/file-saver/test/file-saver-tests.ts:1]: It's not, but it still exposes a `saveAs` export, hence that's why I'm testing it.

Comment (BendingBender) [reply types/file-saver/test/file-saver-tests.ts:1]: And by the way, this is not a default export that I'm testing, a default export would be: `import saveAs from 'file-saver'`.

Comment (JounQin) [reply types/file-saver/test/file-saver-tests.ts:1]: > it still exposes a saveAs export

You mean `saveAs.saveAs`, right? It is still commonjs, not ES `export`.

Comment (BendingBender) [reply types/file-saver/test/file-saver-tests.ts:1]: It is, but it's compatible with ESM at least with TS and WebPack.

**Review by weswigham — APPROVED** [review]

**Comment by BendingBender** [comment]
Ready to merge

**Pull request merged**